hammr/commands/deploy/deploy.py

```diff
@@ -0,0 +1,152 @@
+        doParser = ArgumentParser(prog=self.cmd_name + " terminate", add_help=True,
+                                  description="Terminate a deployment")
+        mandatory = doParser.add_argument_group("mandatory arguments")
+        mandatory.add_argument('--id', dest='id', required=True,
+                               help="id of the deployment to terminate")
+        optional = doParser.add_argument_group("optional arguments")
+        optional.add_argument('--force', '-f', dest='force', required=False, action='store_true', help='Terminate the deployment without asking for confirmation')
+        return doParser
+
+    def do_terminate(self, args):
+        try:
+            # add arguments
+            doParser = self.arg_terminate()
+            doArgs = doParser.parse_args(shlex.split(args))
+
+            if (not doArgs.force and generics_utils.query_yes_no("Do you really want to delete deployment with id '" + str(doArgs.id) + "'")) or doArgs.force:
+                status = self.api.Users(self.login).Deployments(doArgs.id).Status.Getdeploystatus()
+                deployment_terminate = self.api.Users(self.login).Deployments(doArgs.id).Terminate()
+                printer.out("Deployment is stopping")
+                bar = ProgressBar(widgets=[BouncingBar()], maxval=UnknownLength)
+                bar.start()
+                i = 1
+                while (self.get_deploy(doArgs.id)):
+                    if status.message != "on-fire":
+                        status = self.api.Users(self.login).Deployments(doArgs.id).Status.Getdeploystatus()
+                        if status.message == "on-fire":
+                            break
+                    time.sleep(1)
+                    bar.update(i)
+                    i += 2
+                bar.finish()
+
+                if self.get_deploy(doArgs.id):
+                    printer.out("Could not terminate the deployment.", printer.ERROR)
+                    if status.message == "on-fire" and status.detailedError:
+                        printer.out(status.detailedErrorMsg, printer.ERROR)
+                    return 1
+                else:
+                    printer.out("Deployment terminated", printer.OK)
+
+            return 0
+        except ArgumentParserError as e:
+            printer.out("ERROR: In Arguments: " + str(e), printer.ERROR)
+            self.help_terminate()
+        except Exception as e:
+            return handle_uforge_exception(e)
+
+    def help_terminate(self):
+        doParser = self.arg_terminate()
+        doParser.print_help()
+
+    def get_deploy(self, deploy_id):
+        deployments = self.api.Users(self.login).Deployments.Getall()
+        deployments = deployments.deployments.deployment
+
+        if deployments is None or len(deployments) == 0:
+            return False
+        else:
+            for deployment in deployments:
+                deployment_id = deployment.applicationId
+                if deployment_id == deploy_id:
```

Comment:
to avoid confusion between deployment_id and deploy_id, deploy_id may be renamed "searched_deployment_id" for example.